processing/src/main/java/org/apache/druid/query/expression/LookupExprMacro.java

```diff
@@ -116,4 +126,15 @@ public void decorateCacheKeyBuilder(CacheKeyBuilder builder)
 
     return new LookupExpr(arg);
   }
+
+  private String getReplaceMissingValueWith(final List<Expr> args)
+  {
+    if (args.size() > 2) {
+      final Expr missingValExpr = args.get(2);
+      if (missingValExpr.isLiteral()) {
+        return missingValExpr.getLiteralValue().toString();
```

Comment:
Q. when used in sqlCompatibleMode if there is a value like empty string should it be converted to null ?

Reply:
Nope, regardless of the sqlCompatibleMode, replaceMissingValueWith is literal value passed by user and feature expect to replace it as it is.